lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -158,10 +160,25 @@ class ConvertAtenMmOp : public OpConversionPattern<AtenMmOp> {
                        loc, zeroFill.getType(), ValueRange{lhs, rhs}, zeroFill)
                    .getResult(0);
     } else {
+      auto accumulatorTensorType = getDefaultAccType(rewriter, resultDTy);
```

Comment:
This should not be overwritten. You should be checking the determining the accumulator type above `createZeroInitTensor` then pass it into the zero init work. Doing so will drastically simplify the change as `MatmulUnsignedOp`, `QuantizedMatmulOp`, and `MatmulOp` should all consider accumulator checks.

Reply:
That makes sense